armory/data/utils.py

```diff
@@ -92,6 +92,16 @@ def download_verify_dataset_cache(dataset_dir, checksum_file, name):
     cache_dir = os.path.join(dataset_dir, "cache")
     os.makedirs(cache_dir, exist_ok=True)
     tar_filepath = os.path.join(cache_dir, os.path.basename(s3_key))
+    if os.path.exists(tar_filepath):
+        # Check existing download to avoid falling back to processing data
+        logger.info(f"{tar_filepath} exists. Verifying...")
+        try:
+            verify_size(tar_filepath, int(file_length))
+            verify_sha256(tar_filepath, hash)
```

Comment:
If tar_filepath exists and has a valid hash, the sha256 is verified twice. Not a hugely expensive operation, but might be worth having a boolean variable that tracks whether the first verification succeeded.

Reply:
Good call

Reply:
Should work now.
```
>>> d = datasets.librispeech_dev_clean("train", 1, 1)
2020-03-26 20:51:29 5315d24e3726 armory.data.utils[14] INFO /armory/datasets/cache/librispeech_dev_clean_split.tar.gz exists. Verifying...
2020-03-26 20:51:34 5315d24e3726 armory.data.utils[14] INFO Dataset already downloaded.
2020-03-26 20:51:34 5315d24e3726 armory.data.utils[14] INFO Extracting .tfrecord files from download...
2020-03-26 20:52:53 5315d24e3726 absl[14] INFO No config specified, defaulting to first: librispeech_dev_clean_split/plain_text
2020-03-26 20:52:53 5315d24e3726 absl[14] INFO Overwrite dataset info from restored data version.
2020-03-26 20:52:53 5315d24e3726 absl[14] INFO Reusing dataset librispeech_dev_clean_split (/armory/datasets/librispeech_dev_clean_split/plain_text/1.1.0)
2020-03-26 20:52:53 5315d24e3726 absl[14] INFO Constructing tf.data.Dataset for split train, from /armory/datasets/librispeech_dev_clean_split/plain_text/1.1.0
```